src/pages/Signup/Signup.js

```diff
@@ -1,5 +1,253 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
+import './Signup.scss';
+import TERMS_DATA from './SignupData';
 
 export default function Signup() {
-  return <div>hi</div>;
+  const [memberInput, setMemberInput] = useState({
+    email: '',
+    password: '',
+    passwordRe: '',
+    address: '',
+    isVegan: false,
+  });
+  const [emailValid, setEmailValid] = useState(false);
+  const [passwordValid, setPasswordValid] = useState(false);
+  const [passwordReValid, setPasswordReValid] = useState(false);
+  const [addressValid, setAddressValid] = useState(false);
+  const [checkedList, setCheckedList] = useState([]);
+
+  //전체 체크
+  const onCheckedAll = useCallback(checked => {
+    if (checked) {
+      const checkedListArray = [];
+
+      TERMS_DATA.forEach(term => checkedListArray.push(term.id));
+      setCheckedList(checkedListArray);
+    } else {
+      setCheckedList([]);
+    }
```

Comment:
```suggestion
      const newCheckedList = checked ? TERMS_DATA.map(term => term.id) : [];
      setCheckedList(newCheckedList);
    }
```
- 새로운 배열을 만들때는 forEach 보다는 map이 적합합니다.
- setChecked 부분은 계속 반복되고 checked값에 따라서 인자로 들어갈 value만 바뀌면 되기때문에 삼항연산자로 표현하는게 적합해보이네요